pyspeckit/spectrum/fitters.py

```diff
@@ -412,14 +412,25 @@ def EQW(self, plot=False, plotcolor='g', fitted=True, continuum=None,
         if plot and self.Spectrum.plotter.axis:
             if midpt_location == 'plot-center':
                 midpt_pixel = np.round((xmin+xmax)/2.0)
-                midpt       = self.Spectrum.xarr[midpt_pixel]
+                midpt       = self.Spectrum.xarr[midpt_pixel].value
             elif midpt_location == 'fitted':
                 try:
-                    midpt = sp.specfit.parinfo.SHIFT0.value
+                    PI_keys = self.Spectrum.specfit.parinfo.keys()
+                    shifts = [self.Spectrum.specfit.parinfo[PI_keys[x]].value for x in range(1, len(PI_keys), 3)]
                 except AttributeError:
                     raise AttributeError("Can only specify midpt_location="
                                          "fitted if there is a SHIFT parameter"
                                          "for the fitted model")
+                # We choose to display the eqw fit at the center of the fitted line set,
+                # closest to the passed window.
+                # Note that this has the potential to show a eqw "rectangle" centered
+                # on a fitted line other than the one measured for the eqw call, if
+                # there are more than one fitted lines within the window.
+                midpt_pixel = (xmin+xmax)/2
+                midval = self.Spectrum.xarr[midpt_pixel].value
+                shifts.sort(key=lambda s: abs(s-midval))
+                midpt = shifts[0]
```

Comment:
Looks fine, but could also do:
```
midpt_index = np.argmin(np.abs(shifts-midval))
midpt = shifts[midpt_index]
```
I'm not sure if it's faster, but it seems a little clearer to me.

I think the logic is fine.